### ISSUE TITLE
Use opaque types instead of handle enums

### ISSUE THREAD

**Issue by anyputer** [Snektron/vulkan-zig#115]
Instead of the following generated code:
```zig
pub const Instance = enum(usize) { null_handle = 0, _ };
```
we can make use of opaque types:
```zig
pub const Instance = opaque {}; // maybe *opaque {}?
```
This improves type safety with explicit nullability in generated code, and makes it easier to use with other C APIs that expect an opaque pointer. For example the instance argument in `destroyInstance` is not obviously nullable until you read the Vulkan spec, and `null` is obviously cleaner than `.null_handle`. This seems like a missed opportunity.

**Comment by Snektron** [comment]
Indeed, these were not a thing when vulkan-zig was initially made. Seems like a good idea, though there will be a slight incompatibility between dispatchable and non-dispatchable handles. I don't think that is much of a problem, though.

**Comment by VoilaNeighbor** [comment]
For most of the times it's not a problem. Just when interfacing with other C libraries we have to do this:

```zig
@ptrFromInt(@intFromEnum(handle))
```
